src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/IProjectSnapshotExtensions.cs

```diff
@@ -15,7 +15,7 @@ namespace Microsoft.CodeAnalysis.Razor.ProjectSystem;
 
 internal static class IProjectSnapshotExtensions
 {
-    public static RazorProjectInfo ToRazorProjectInfo(this IProjectSnapshot project, string serializedFilePath)
+    public static RazorProjectInfo ToRazorProjectInfo(this IProjectSnapshot project, string? serializedFilePath = null)
```

Comment:
Perhaps as an alternative to making this nullable, perhaps it would be safer to just always create the project info with `serializedFilePath: project.Key.Id`?

Reply:
I don't like the idea of having a `RazorProjectInfo.SerializedFilePath` value that points to a nonsense path. I'd rather just increment the serialization version for `RazorProjectInfo` and add `ProjectKey`.

Reply:
I went ahead and did this work, which required re-layering `ProjectKey` a bit. Note that this PR now updates the serialization format, so we'll need a dual insertion with Roslyn for C# Dev Kit.